dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -334,6 +334,19 @@ ClassPathEntry findClassLocation(String className) {
     return Iterables.getFirst(classFileNameToClassPathEntry.get(specialLocation), null);
   }
 
+  /**
+   * Returns class file name which is a location of a class file within a JAR file. Usually it is
+   * the same as {@code classBinaryName}, however it may be different if the JAR file has framework-
+   * specific prefix such as "WEB-INF.classes".
+   */
+  String getClassFileName(String classBinaryName) {
```

Comment:
This logic belongs in ClassPathEntry and ClassFile, not here.

Reply:
ClassPathEntry cannot convert class binary names to class file names, because it does not read the content of .class files. On the other hand, ClassDumper (through classRepository) already reads the content of class files and thus can convert class binary names to class file names.

Reply:
How is BCEL getting these names? Is it reading the individual .class files? Or does it know something we don't know about the format of JAR files?

Reply:
BCEL's JavaClass reads class name from class file's constant pool: https://github.com/apache/commons-bcel/blob/master/src/main/java/org/apache/bcel/classfile/JavaClass.java#L157

Reply:
It feels right that readClassFileNames be the place this happens, if necessary by reading the byte code, though perhaps there's a simpler way to do this. For instance we could strip off known prefixes.

Reply:
As the classRepository maintains the mappings, we do not need to read the class files again in ClassPathEntry.